e2e/create-cluster_test.go

```diff
@@ -634,4 +634,53 @@ var _ = Describe("CreateCluster", func() {
 		By("Expecting both KubevirtMachines stabilize to a ready=true again.")
 		waitForMachineReadiness(2, 0)
 	})
+
+	// This test will create a tenant cluster from `templates/cluster-template-ext-infra.yaml` template.
+	// 1. generate secret in 'capk-system' namespace, containing external infra kubeconfig and namespace data
+	// (note, the kubeconfig actually points to the same cluster, which is fine for the sake of the test)
+	// 2. generate tenant cluster yaml manifests with `clusterctl generate cluster` command
+	// 3. apply generated tenant cluster yaml manifests
+	// 4. verify that tenant cluster machines booted and bootstrapped
+	// 5. verify that tenant cluster control plane came up successfully
+	It("Create a simple tenant cluster on external infrastructure", func() {
+		By("generating a secret with external infrastructure kubeconfig and namespace")
+		kubeconfig, err := os.ReadFile(os.Getenv("KUBECONFIG"))
```

Comment:
@nirarg  you're right, we're kind of test external infra feature with `clusterkubevirtadm` already, but I thought having an explicit test code in e2e for this can also help to "document" it in tests as well, mainly for reference for developers. wdyt?

Reply:
I agree that it will be good to have this e2e explicit test
I think that also using `clusterkubevirtadm` in this test can be helpful

Reply:
i think this test is a good start that proves that our controllers work with external infra. we should follow up by integrating clusterkubevirtadm into this test at some point... 

so basically, i think this test has value now, and we can increase the value with exercising clusterkubevirtadm later on